### PR TITLE
Change the default reference configurations

### PR DESCRIPTION
**Pull request by solangegbv**
to align every rod with the x-axis by default.

This change includes:
- Adjusting the default reference configurations
- Changing the documentation to reflect the new default behavior
- Changing the default stiffness matrix to match the new reference configurations

All previous "equilibrium" and "rest" are now called "reference".

### CODE REVIEW

File: src/soromox/symbolic_derivation/planar_hsa.py [hunk 1]
```diff
@@ -544,9 +544,9 @@ def symbolically_derive_planar_hsa_model(
             "rout": rout_syms,
             "rin": rin_syms,
             "roff": roff_syms,
-            "kappa_b_eq": kappa_b_eq_syms,
```
Comment (mstoelzle) [inline src/soromox/symbolic_derivation/planar_hsa.py:547]: If you do these changes to the symbolic derivation, you will also have to generate the symbolic expressions and push them in this PR.

File: tests/test_planar_pcs_num.py [hunk 2]
```diff
@@ -72,7 +72,7 @@ def test_planar_cs_num(
     # activate all strains (i.e. bending, shear, and axial)
     strain_selector = jnp.ones((3,), dtype=bool)
 
-    xi_eq = jnp.array([0.0, 0.0, 1.0])
+    xi_ref = jnp.array([0.0, 0.0, 1.0])
```
Comment (mstoelzle) [inline tests/test_planar_pcs_num.py]: This `xi_ref` should probably be changed to `jnp.array([0.0, 1.0, 0.0])`?

File: src/soromox/systems/planar_pcs.py [hunk 3]
```diff
@@ -344,46 +355,101 @@ def update_params(self, params: Dict[str, Array]) -> "PlanarPCS":
...
+                - "D": List/Array of (num_segments x num_segments) floats
+                    Damping matrix of each segment [Pa*s]
         """
         # Apply updates sequentially
         updated_self = self
         
         if "th0" in params:
-            updated_self = eqx.tree_at(lambda m: m.th0, updated_self, jnp.asarray(params["th0"], dtype=jnp.float64))
+            th0 = params["th0"]
+            if not (isinstance(th0, (float, int, jnp.ndarray))):
```
Comment (mstoelzle) [inline src/soromox/systems/planar_pcs.py:381]: Having conditions in the `update_params` method is potentially problematic, as this function is designed to be used in jitted settings. We can try though.

File: examples/simulate_planar_pcs.py [hunk 4]
```diff
@@ -153,7 +152,7 @@ def update_plot(frame_idx):
         "L": 1e-1 * jnp.ones((num_segments,)),
         "r": 2e-2 * jnp.ones((num_segments,)),
         "rho": rho,
-        "g": jnp.array([0.0, -9.81]),
+        "g": jnp.array([0.0, 9.81]), # gravity vector [m/s^2] UP!
```
Comment (mstoelzle) [inline examples/simulate_planar_pcs.py:155]: So I assume that here gravity is pointing upwards?

File: src/soromox/symbolic_expressions/pendulum_nl-1.dill
Comment (mstoelzle) [inline src/soromox/symbolic_expressions/pendulum_nl-1.dill:1]: We need to keep the pendulum symbolic expressions I think, no? Or are you planning to also do a numeric implementation of the dynamics there?

File: src/soromox/symbolic_expressions/planar_hsa_ns-1_nrs-2.dill
Comment (mstoelzle) [inline src/soromox/symbolic_expressions/planar_hsa_ns-1_nrs-2.dill:1]: We need to keep the planar HSA symbolic expressions I think, no? Or are you planning to also do a numeric implementation of the dynamics there?

**Review by mstoelzle — CHANGES_REQUESTED** [review]

File: examples/simulate_planar_pcs.py (hunk 4 above)
Comment (solangegbv) [reply examples/simulate_planar_pcs.py:155]: Actually I think it was the case on your previous examples so I putted that same way. I'm not sure why, but I just imitated your previous example.

File: src/soromox/symbolic_expressions/pendulum_nl-1.dill
Comment (solangegbv) [reply src/soromox/symbolic_expressions/pendulum_nl-1.dill:1]: No, for the moment I'm keeping them, I'll see if I eventually have the time.

File: src/soromox/symbolic_expressions/planar_hsa_ns-1_nrs-2.dill
Comment (solangegbv) [reply src/soromox/symbolic_expressions/planar_hsa_ns-1_nrs-2.dill:1]: No, for the moment I'm keeping them, I'll see if I eventually have the time.

**Review by mstoelzle — APPROVED** [review]
